src/Card.js

```diff
@@ -0,0 +1,313 @@
+      this.on('dragstart', this.dragStart)
+      this.on('drag', (ptr,dragX,dragY)=>this.drag(ptr, dragX, dragY));
+      this.on('dragend', this.dragEnd);
+      this.on('pointerover', (ptr,localX,localY)=>this.hover(ptr,localX,localY))
+      this.on('pointerout', (ptr)=>this.unhover(ptr))
+
+      this.stackCounter.on('dragstart', ()=> this.dragStart())
+      this.stackCounter.on('drag', (ptr,dragX,dragY)=> this.drag(ptr, ptr.worldX + 32.5, ptr.worldY + 77.5, true));
+      this.stackCounter.on('dragend', () => this.dragEnd());
+
+      const buttonArray = [this.rotateButton, this.flipButton, this.shuffleButton, this.stackCounter]
+      buttonArray.forEach( button => {
+        button.on('pointerover', (ptr,localX,localY)=>this.hover(ptr,localX,localY));
+        button.on('pointerout', (ptr)=>this.unhover(ptr));
+      })
+
+      this.rotateButton.on('drag', (ptr,dragX,dragY)=>this.spin(ptr, dragX, dragY));
+      this.rotateButton.on('dragend', (ptr)=>this.dragEnd(ptr));
+      this.flipButton.on('pointerdown',()=>this.startFlip())
+      this.flipButton.on('pointerup',()=>this.flip())
+
+    //card status variables
+    this.cardNumber = cardNumber;
+    this.startFlipClickedDown = false;
+    this.revealed = false;
+    this.spinning = false;
+    this.otherPlayerDragging = false;
+    this.addToHand = false;
+    this.inHand = false;
+    this.playerPickedUp = false;
+    //this.stackNumber should always be the number of the card at the bottom of a given stack.
+    //I.e., card 5 is stacked on card 18, and card 19 is stacked on card 5, all 3 cards will have stackNumber of 18.
+    this.stackNumber = cardNumber
+    //stackOrder is the number of cards from the bottom.
+    //in example above, card18.stackOrder = 1, card5.stackOrder = 2, card19.stackOrder = 3.
+    this.stackOrder = 1
+
+    //overwrite phaser's bullshit toJSON implementation
+    this.toJSON = () => {
+      return {
+        cardNumber: this.cardNumber,
+        x: this.x,
+        y: this.y,
+        rotation: this.rotation,
+        revealed: this.revealed,
+        velocity: this.body.velocity,
+        otherPlayerDragging: this.otherPlayerDragging,
+        inHand: this.inHand,
+        player: this.playerNumber,
+        stackNumber: this.stackNumber,
+        stackOrder: this.stackOrder,
+        depth: this.depth
+      }
+    }
+  }
+
+  getCardsInStack(stackNumber = this.stackNumber) {
+    return Object.keys(this.gameState.cards)
+    .map( key => this.gameState.cards[key])
+    .filter( card => card.stackNumber === stackNumber)
+    .sort( (a,b) => a.stackOrder - b.stackOrder)
+  }
+
+  giveNextStackNumber(stackNumber = this.stackNumber) {
+    return this.getCardsInStack(stackNumber).length + 1
+  }
+
+  getClosestCardNotInStack() {
+    const roomCards = Object.keys(this.gameState.cards).filter(key => Number(key) !== this.cardNumber && this.gameState.cards[key].stackNumber !== this.stackNumber).map( key => this.gameState.cards[key])
```

Comment:
This line is too wide, I'd break it off onto multiple lines like in getCardsInStack